java/src/com/ctre/phoenix/motorcontrol/can/WPI_TalonSRX.java

```diff
@@ -43,10 +45,20 @@ public WPI_TalonSRX(int deviceNumber) {
 	@Override
 	public void set(double speed) {
 		_speed = speed;
-		set(ControlMode.PercentOutput, _speed);
+		set(defaultControlMode, _speed);
 		_safetyHelper.feed();
 	}
-
+	
+	public void setDefaultPercentOutputMode(){
```

Comment:
Why are you calling it `setDefault`. You aren't changing the default. You're changing the current one

Reply:
It's the default *if* nothing is specified

Reply:
you can run `set(ControlMode.Velocity, 3.4)` and then run `set(1.0)`... the latter will not use the velocity mode, unless it's previously been set using this... to do that, you'd need to handle this on the other set methods as well.

Reply:
I'm concerned that opening the set method up to other control modes beyond these, such as position, can be used incorrectly easily.

This implementation allows for it to be easily adapted for anywhere you'd use a standard SpeedController, not sure how you'd map that -1...1 with position..

Reply:
Couldn’t you just have a max position? Not as useful on a spring loaded joystick but on a throttle you could. And current can have a max current easily.

Reply:
so support position/current like this and fallback to vBus % for when someone tries to use motion profiles/magic from WPI wrappers?

Reply:
Another solution might be to have a WPI_ControlMode that includes PercentOutput and Velocity.  So WPI_Talon and WPI_Vic would have setters for the WPI_ControlMode and scalar.  As new control modes are supported - if they make sense - then they are added to WPI_ControlMode.  Underneath the WPI_* class can call the appropriate native control mode.   It's important that students can't code something up can't possibly work.  I think fallback to PercentOutput would be problematic.

Or another solution would be to extend on WPILIB's side so that if you pass CTRE motor controllers, you get more bells and whistles.

Reply:
yeah, fair point, I think `WPI_ControlMode` would work well, and you'd still be able to use the native SRX set methods to control as well right?

Say, DifferentialDrive is set to use two SRXs in velocity mode, but the SRX classes can be used to override this using `set(ControlMode mode, double value)` to use position mode.

Reply:
Although I've heard from many teams planning to use velocity mode for DifferentialDrive, I think extending WPILib drivetrain to support CTRE motor controllers will likely cause confusion with CTRE's existing drivetrains utils.

Reply:
@ozrien what was the reasoning for removing changeControlMode and just passing in the control mode to set?

Reply:
"...SRX set methods to control as well right?" yep.

"...control mode to set".  The single-param set() was confusing.  Teams would changeMode in one spot, and set() way somewhere else .  Then they would modify the changeMode, forgot about the set, then ask us why it doesn't work.    Also if you changeMode to change the mode and don't call set, the motor controller would disable momentarily until the next set() call, because it doesn't know what to do.   It's a strange contract and makes supporting the product difficult. 

Also the mode is what determines the units of the set.  It's dangerous to allow teams to set the "Target" of whatever and not enforce that they instruct the motor controller what mode to do.

Reply:
Ok some people were complaining and I told them it made it easier because you would know exactly what mode its in at each set call.  IMO its better this way